packages/javascript/bh-shared-ui/src/components/CreateUserForm/CreateUserForm.tsx

```diff
@@ -14,426 +14,699 @@
-import React, { useEffect } from 'react';
-import { Controller, useForm } from 'react-hook-form';
+    SelectContent,
+    SelectItem,
+    SelectPortal,
+    SelectTrigger,
+    SelectValue,
+    Tooltip,
+} from '@bloodhoundenterprise/doodleui';
+import { faSearch } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { Alert } from '@mui/material';
+import { CreateUserRequest, Environment, EnvironmentRequest, Role, SSOProvider } from 'js-client-library';
+import { Minus } from 'lucide-react';
+import React, { useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
 import { useQuery } from 'react-query';
 import { MAX_EMAIL_LENGTH, MAX_NAME_LENGTH, MIN_NAME_LENGTH } from '../../constants';
-import { useListDisplayRoles } from '../../hooks';
+import { useAvailableEnvironments } from '../../hooks/useAvailableEnvironments/useAvailableEnvironments';
 import { apiClient } from '../../utils';
 
-export type CreateUserRequestForm = Omit<CreateUserRequest, 'SSOProviderId'> & { SSOProviderId: string | undefined };
+export type CreateUserRequestForm = Omit<CreateUserRequest, 'sso_provider_id'> & {
+    sso_provider_id: string | undefined;
+};
 
 const CreateUserForm: React.FC<{
-    onCancel: () => void;
-    onSubmit: (user: CreateUserRequestForm) => void;
-    isLoading: boolean;
     error: any;
-}> = ({ onCancel, onSubmit, isLoading, error }) => {
-    const {
-        control,
-        handleSubmit,
-        setValue,
-        formState: { errors },
-        setError,
-    } = useForm<CreateUserRequestForm>({
-        defaultValues: {
-            emailAddress: '',
-            principal: '',
-            firstName: '',
-            lastName: '',
-            password: '',
-            needsPasswordReset: false,
-            roles: [1],
-            SSOProviderId: '',
+    isLoading: boolean;
+    onSubmit: (user: CreateUserRequestForm) => void;
+    showEnvironmentAccessControls?: boolean;
+}> = ({ error, isLoading, onSubmit, showEnvironmentAccessControls }) => {
+    const defaultValues = {
+        email_address: '',
+        principal: '',
+        first_name: '',
+        last_name: '',
+        password: '',
+        needs_password_reset: false,
+        roles: [3],
+        sso_provider_id: '',
+        all_environments: false,
+        environment_targeted_access_control: {
+            environments: null,
         },
-    });
+    };
+
+    const form = useForm<CreateUserRequestForm>({ defaultValues });
 
     const [authenticationMethod, setAuthenticationMethod] = React.useState<string>('password');
+    const [needsPasswordReset, setNeedsPasswordReset] = useState(false);
+    const [searchInput, setSearchInput] = useState<string>('');
+    const [selectedEnvironments, setSelectedEnvironments] = useState<string[]>([]);
+    const [selectedRoleValue, setSelectedRoleValue] = useState([3]);
```

Comment:
same comment as this one: https://github.com/SpecterOps/BloodHound/pull/1858/files#r2475736429